aws/resource_aws_ec2_fleet.go

```diff
@@ -190,6 +190,44 @@ func resourceAwsEc2Fleet() *schema.Resource {
 							Default:      1,
 							ValidateFunc: validation.IntAtLeast(1),
 						},
+						"maintenance_strategies": {
+							Type:     schema.TypeList,
+							Optional: true,
+							MaxItems: 1,
+							DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+								if old == "1" && new == "0" {
+									return true
+								}
+								return false
+							},
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"capacity_rebalance": {
+										Type:     schema.TypeList,
+										Optional: true,
+										MaxItems: 1,
+										DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+											if old == "1" && new == "0" {
```

Comment:
Want to confirm here that the idea is that if `maintenance_strategies` or `capacity_rebalance` is removed from the configuration, that should be ignored (not removed through the API)?

Reply:
I would defer to the maintainers to determine the behavior that would be considered expected. If configurations are normally removed, is the user expectation to reset them?

Reply:
I copied the logic that I saw used for spot_options and on_demand_options. I am open to suggestions on the best way to handle this.

Reply:
@bill-rich would you be willing to also evaluate the spot_maintenance_strategies definition in the resource_aws_spot_fleet_request.go ?

Reply:
@AdamTylerLynch Some context on this type of diff suppression (see #7180):

```go
// suppressMissingOptionalConfigurationBlock handles configuration block attributes in the following scenario:
//  * The resource schema includes an optional configuration block with defaults
//  * The API response includes those defaults to refresh into the Terraform state
//  * The operator's configuration omits the optional configuration block
```